contracts/ligo/src/swaps/README.md

```diff
@@ -157,3 +172,55 @@ Be sure to handle not only the errors listed in the entrypoints description, but
 
 1. Forbid empty offered/requested assets to avoid user's mistakes.
 2. Preserve info about resolved swaps in storage - decided not to go with this is favor of using blockchain explorers.
+
+## Whitelisted extension
+
+This contract additionally provides a way to restrict the set of allowed FA2 tokens that can participate in exchanges.
+This set is managed by the administrator entity.
+
+Source code: [ligo](fa2_whitelisted_swap.mligo).
+
+### Contract parts
+
+The plan is to develop one contract consisting of a few components that can be reused for other purposes.
+
+The contract has to consist of the following parts:
+1. Base swap contract;
+2. Simple administrator contract. We picked the existing [`simple_admin` contract](../../fa2_modules/admin/non_pausable_simple_admin.mligo) that is already present in the `minter-sdk` repository.
+3. Whitelisting component (reuses the swaps contract and implements the whitelisting capabilities).
+
+The last point will be described further in this document.
+
+<a name="whitelist-types"></a>
+### Types
+
+The storage of the whitelist component is solely `big_map address ()`.
+
+<a name="whitelist-entrypoints"></a>
+### Entrypoints
+
+#### Update whitelist
+
+```ocaml
+| Update_allowed of (address, unit) big_map
+```
+
+This entrypoint allows setting a new whitelist, overriding the current one.
+
```

Comment:
As I was saying in the other thread I think we should add a simple line here to mention why we get a `big_map`

Reply:
Yeah, right.